ano/commands/build.py

```diff
@@ -259,6 +266,23 @@ def _scan_dependencies(self, dirName, lib_dirs, inc_flags):
 
         return used_libs
 
+    def check_memory(self, args):
+        board = self.e.board_model(args.board_model)
+        boardVariant = args.cpu if ('cpu' in args) else None;
+        maximum_size = int(BoardModels.getValueForVariant(board, boardVariant,
+            'upload', 'maximum_size'))
+        maximum_data = int(BoardModels.getValueForVariant(board, boardVariant,
```

Comment:
maximum_data_size appears to be either newer or optional. Can you handle the KeyError thrown from this line for older IDE versions?